ShadowApplication receiverInfo未判空

![image](https://user-images.githubusercontent.com/19511418/148347965-7ee92c4f-44f0-4884-8dcc-4be321e4f001.png)
报错:
`2022-01-06 15:50:01.525 9111-9136/? E/JavaBinder: *** Uncaught remote exception!  (Exceptions are not yet supported across processes.)

    java.util.concurrent.ExecutionException: java.util.concurrent.ExecutionException: com.tencent.shadow.core.loader.exceptions.CreateApplicationException: java.lang.NullPointerException: storage == null
        at java.util.concurrent.FutureTask.report(FutureTask.java:93)
        at java.util.concurrent.FutureTask.get(FutureTask.java:163)
        at com.tencent.shadow.dynamic.loader.impl.DynamicPluginLoader.loadPlugin(DynamicPluginLoader.kt:82)
        at com.tencent.shadow.dynamic.loader.impl.PluginLoaderBinder.onTransact(PluginLoaderBinder.kt:43)
        at android.os.Binder.execTransact(Binder.java:446)
     Caused by: java.util.concurrent.ExecutionException: com.tencent.shadow.core.loader.exceptions.CreateApplicationException: java.lang.NullPointerException: storage == null
        at java.util.concurrent.FutureTask.report(FutureTask.java:93)
        at java.util.concurrent.FutureTask.get(FutureTask.java:163)
        at com.tencent.shadow.core.loader.blocs.LoadPluginBloc$loadPlugin$buildRunningPlugin$1.run(LoadPluginBloc.kt:127)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
        at java.lang.Thread.run(Thread.java:818)
     Caused by: com.tencent.shadow.core.loader.exceptions.CreateApplicationException: java.lang.NullPointerException: storage == null
        at com.tencent.shadow.core.loader.blocs.CreateApplicationBloc.createShadowApplication(CreateApplicationBloc.kt:69)
        at com.tencent.shadow.core.loader.blocs.LoadPluginBloc$loadPlugin$buildApplication$1.call(LoadPluginBloc.kt:108)
        at com.tencent.shadow.core.loader.blocs.LoadPluginBloc$loadPlugin$buildApplication$1.call(LoadPluginBloc.kt:101)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
        at java.lang.Thread.run(Thread.java:818) 
     Caused by: java.lang.NullPointerException: storage == null
        at java.util.Arrays$ArrayList.<init>(Arrays.java:38)
        at java.util.Arrays.asList(Arrays.java:155)
        at com.tencent.shadow.core.runtime.ShadowApplication.setBroadcasts(ShadowApplication.java:157) <----看这里!!!
        at com.tencent.shadow.core.loader.blocs.CreateApplicationBloc.createShadowApplication(CreateApplicationBloc.kt:58)
        at com.tencent.shadow.core.loader.blocs.LoadPluginBloc$loadPlugin$buildApplication$1.call(LoadPluginBloc.kt:108) 
        at com.tencent.shadow.core.loader.blocs.LoadPluginBloc$loadPlugin$buildApplication$1.call(LoadPluginBloc.kt:101) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:237) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587) 
        at java.lang.Thread.run(Thread.java:818) `

--- a/projects/sdk/core/runtime/src/main/java/com/tencent/shadow/core/runtime/ShadowApplication.java
+++ b/projects/sdk/core/runtime/src/main/java/com/tencent/shadow/core/runtime/ShadowApplication.java
@@ -29,9 +29,7 @@ import android.content.IntentFilter;
 import android.content.res.Configuration;
 import android.os.Build;
 
-import java.util.Arrays;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
 /**
@@ -41,7 +39,7 @@ public class ShadowApplication extends ShadowContext {
 
     private Application mHostApplication;
 
-    private Map<String, List<String>> mBroadcasts;
+    private Map<String, String[]> mBroadcasts;
 
     private ShadowAppComponentFactory mAppComponentFactory;
 
@@ -68,16 +66,19 @@ public class ShadowApplication extends ShadowContext {
 
         isCallOnCreate = true;
 
-        for (Map.Entry<String, List<String>> entry : mBroadcasts.entrySet()) {
+        for (Map.Entry<String, String[]> entry : mBroadcasts.entrySet()) {
             try {
-                Class<?> clazz = mPluginClassLoader.loadClass(entry.getKey());
+                String receiverClassname = entry.getKey();
+                Class<?> clazz = mPluginClassLoader.loadClass(receiverClassname);
                 BroadcastReceiver receiver = ((BroadcastReceiver) clazz.newInstance());
-                mAppComponentFactory.instantiateReceiver(mPluginClassLoader, entry.getKey(), null);
+                mAppComponentFactory.instantiateReceiver(mPluginClassLoader, receiverClassname, null);
 
                 IntentFilter intentFilter = new IntentFilter();
-                for (String action:entry.getValue()
-                     ) {
-                    intentFilter.addAction(action);
+                String[] receiverActions = entry.getValue();
+                if (receiverActions != null) {
+                    for (String action : receiverActions) {
+                        intentFilter.addAction(action);
+                    }
                 }
                 registerReceiver(receiver, intentFilter);
             } catch (Exception e) {
@@ -151,10 +152,10 @@ public class ShadowApplication extends ShadowContext {
     }
 
     public void setBroadcasts(PluginManifest.ReceiverInfo[] receiverInfos) {
-        Map<String, List<String>> classNameToActions = new HashMap<>();
+        Map<String, String[]> classNameToActions = new HashMap<>();
         if (receiverInfos != null) {
             for (PluginManifest.ReceiverInfo receiverInfo : receiverInfos) {
-                classNameToActions.put(receiverInfo.className, Arrays.asList(receiverInfo.actions));
+                classNameToActions.put(receiverInfo.className, receiverInfo.actions);
             }
         }
         mBroadcasts = classNameToActions;

--- a/projects/test/plugin/general-cases/test-plugin-general-cases/src/main/AndroidManifest.xml
+++ b/projects/test/plugin/general-cases/test-plugin-general-cases/src/main/AndroidManifest.xml
@@ -67,6 +67,7 @@
                 <action android:name="com.tencent.test.normal.action" />
             </intent-filter>
         </receiver>
+        <receiver android:name=".lib.usecases.receiver.ReceiverWithoutAction" />
 
         <activity android:name=".lib.usecases.provider.TestFileProviderActivity" />
 

--- a/projects/test/plugin/general-cases/test-plugin-general-cases/src/main/java/com/tencent/shadow/test/plugin/general_cases/lib/usecases/receiver/ReceiverWithoutAction.java
+++ b/projects/test/plugin/general-cases/test-plugin-general-cases/src/main/java/com/tencent/shadow/test/plugin/general_cases/lib/usecases/receiver/ReceiverWithoutAction.java
@@ -1,0 +1,33 @@
+/*
+ * Tencent is pleased to support the open source community by making Tencent Shadow available.
+ * Copyright (C) 2019 THL A29 Limited, a Tencent company.  All rights reserved.
+ *
+ * Licensed under the BSD 3-Clause License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.tencent.shadow.test.plugin.general_cases.lib.usecases.receiver;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+/**
+ * 测试插件中包含无Action的Receiver能否正常启动
+ */
+public class ReceiverWithoutAction extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+    }
+}

